freqtrade/strategy/hyper.py

```diff
@@ -0,0 +1,170 @@
+"""
+IHyperStrategy interface, hyperoptable Parameter class.
+This module defines a base class for auto-hyperoptable strategies.
+"""
+from typing import Iterator, Tuple, Any, Optional, Sequence, Union
+
+from skopt.space import Integer, Real, Categorical
```

Comment:
this will become - uh - complicated now ...

technically, freqtrade doesn't require scikit-optimize (or scikit-learn) to be installed to run basic bot operations (basically everything except hyperopt / hyperopt utils).
This is important as installing scikit-learn on a raspberry is HORROR (compilation takes about a day last i tried).

I don't think that's going to be an issue (just some more code) - we can either wrap the import in a try/except block (catching `ImportError`), and disable parameter optimization for this case, while keeping the "parameter" itself alive (just without the `.get_space()` method).

alternatively, the imports could be moved to the respective "get_space" methods (basically the same approach, but on a different level) - and simply raise an OperationalException when called without having scikit-optimize installed (we don't expect to have `parameter.get_space()` called form non-hyperopt processes i'd think).

---

thinking about it, we should prefer the 2nd option i think.
This would avoid importing skopt when it's not necessary / used (we're not going to use it unless we're hyperopting) - and therefore speed up the startup process for "non-hyperopt" processes.
We'll obviously need to remove the typehints - but i guess we can survive without them for this class...

Reply:
In 58cd40f  i made skopt imports optional by catching import error at module level and using type names as strings in type hints. This gives us best of both worlds - string type name still resolves (at least in intellij IDE) to a type while import is being optional.